drivers/hmis/app/graphql/types/hmis_schema/ce_referral_source_enrollment.rb

```diff
@@ -0,0 +1,132 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Types
+  class HmisSchema::CeReferralSourceEnrollment < Types::BaseObject
+    # object is an OpenStruct with the following shape:
+    # {
+    #   enrollment: Hmis::Hud::Enrollment,
+    #   definition_identifiers: [String],
+    # }
+
+    # Why definition_identifiers?
+    # Often, a community will have a custom assessment that collects eligibility/prioritization info for CE.
+    # We want to show the user, for example, "When was this enrollment's last Housing Needs Assessment"?
+    # when they are looking at a list of candidate enrollments to choose for the referral's source enrollment.
+
+    # Why not use Types::HmisSchema::Enrollment?
+    # This type does NOT have object-level authorization. The user may be able to see possible source enrollments for
+    # a referral, even without permission to see full enrollments in the project.
+
+    field :id, ID, null: false
+
+    field :client_name, String, null: false
+
+    field :project_name, String, null: false
+    field :project_type, Types::HmisSchema::Enums::ProjectType, null: false
+
+    field :entry_date, GraphQL::Types::ISO8601Date, null: false
+    field :exit_date, GraphQL::Types::ISO8601Date, null: true
+    field :in_progress, Boolean, null: false
+    field :auto_exited, Boolean, null: false
+
+    field :relationship_to_ho_h, HmisSchema::Enums::Hud::RelationshipToHoH, null: false, default_value: 99
+    field :household_size, Integer, null: false
+    field :other_household_member_names, [String], null: false
+
+    field :assessments, [Types::HmisSchema::AssessmentNameWithDate], null: false
+
+    def id
+      object.enrollment.id
+    end
+
+    def client_name
+      # Source enrollments could be from different data sources, so return the client name.
+      # TODO(#7591) - ensure that this works correctly
+      client = load_ar_association(object.enrollment, :client)
+      current_permission?(permission: :can_view_client_name, entity: client) ? client.brief_name : client.masked_name
```

Comment:
This will only resolve the name if the client is from the current data source. (That logic is encoded in current_permission?). Just calling that out, maybe add it as a comment, we can probably leave it for now